frontend/src/Editor/DataSourceManager/DataSourceManager.jsx

```diff
@@ -490,7 +494,7 @@ class DataSourceManagerComponent extends React.Component {
       const filteredDatasources = this.state.filteredDatasources.map((datasource) => {
         const src = datasource?.iconFile?.data
           ? `data:image/svg+xml;base64,${datasource.iconFile?.data}`
-          : datasource.kind.toLowerCase();
+          : datasource?.kind?.toLowerCase();
```

Comment:
@vjaris42 Do we need all the optional chaining if we have a state `pluginsLoaded` to see if datasources are present?

Reply:
As discussed with @arpitnath for now we use both as a precaution. Later when we revamp the datasourcemodal we wont have the chaining.